spec/formatting.md

```diff
@@ -29,15 +129,114 @@ the local variable takes precedence.
 It is an error for a local variable definition to
 refer to a local variable that's defined after it in the message.
 
+The resolution of a _variable_ MAY fail if no value is identified for its _name_.
+If this happens, an Unresolved Variable error MUST be emitted.
+If a _variable_ would resolve to a _fallback value_,
+this MUST also be considered a failure.
+
+### Function Resolution
+
+To resolve the value of an _expression_ with a _function_ _annotation_,
+the following steps are taken:
+
+1. If the _expression_ includes an _operand_, resolve its value.
+   If this fails, use a _fallback value_ for the _expression_.
+2. Based on the _function_ starting sigil and _name_,
+   find the appropriate function implementation from the _function registry_.
+   If this fails, emit an Unknown Function error
+   and use a _fallback value_ for the _expression_.
+3. Resolve the _option_ values as a mapping of string identifiers to values.
+   For each _option_, resolve its right-hand side value.
+   If this succeeds, assign it to the mapping with the _name_ of the _option_.
+   If this fails, do not set any value in the resolved mapping for this _option_.
```

Comment:
```suggestion
   For each _option_:
     * If its right-hand side successfully resolves to a value, bind the _name_ of the _option_ to the resolved value in the mapping.
     * Otherwise, do not bind the _name_ of the _option_ to any value in the mapping.
```